tyssue/core/objects.py

```diff
@@ -142,6 +146,11 @@ def edge_df(self, value):
 
     @property
     def cell_df(self):
+        #- BC -#
+        # Recommends to test if the epithelium has a 'cell' df before returning,
+        # some epitheliums don't require this object (2D)
+        # Should it raise a warning/error if the user asks for
+        # a cell_df from an eptm that doesn't have one ?
```

Comment:
Not really... In my view, we should avoid very long backtraces for obvious errors... But in this particular case, the error is risen pretty fast. I'm just not a fan of custom error classes, the ones defined in the standard lib are usually enough.

As a rule of the thumb, I'd say that if you have to scratch your head to understand where the error came from, it's better to catch it earlier.